internal/parse/parts.go

```diff
@@ -0,0 +1,155 @@
+package parse
+
+import "fmt"
+
+// queryPart is an interface to be implemented by all parts.
+type queryPart interface {
+	// String-like output for debugging purposes
+	String() string
+
+	// Translate part to its Sql equivalent
+	ToSql() string
+}
+
+// qualifiedExpression represents a qualified expression
+// qualified. Examples are:
+//	... column as &Person ... 	<--- "column" is unqualified
+//	... t.column as &Person.name	<--- "name" and "column" and qualified
+type qualifiedExpression struct {
+	qualifier string
+	name      string
+}
+
+// InputPart represents an expression used as input in the SDL query.
+// Examples:
+//	$Address
+//	$Address.postal_code
+type InputPart struct {
+	goType
+}
+
+// NewInputPart is a constructor that returns a pointer to an InputPart.
+func NewInputPart(typeName string, tagName string) (*InputPart, error) {
+	if typeName == "" && tagName != "" {
+		return nil, fmt.Errorf("malformed InputPart")
+	}
+	return &InputPart{
+		goType{
+			qualifiedExpression{qualifier: typeName,
+				name: tagName,
+			},
+		},
+	}, nil
```

Comment:
That's quite a bit of syntax to replace what seems trivial:
```
&InputPart{
        Type: "Foo",
        Name: "bar",
}
```
Why the additional complexity? We don't need the error checking here as we can do it when processing the query.

Also, this is smelling slightly a bit like a name soup, in the sense we have several different ways to refer to the same data. For example, a Go field is now a "name", a "tag", and an "expression", that sits in a "type", or a "qualifier". We'll likely benefit from simplifying this: it's a type name, with a field name.

just here we have three different ways to name the same thing: tagName,